spec/requests/generic_oauth_index_spec.rb

```diff
@@ -116,11 +116,12 @@ def validate_json(scope, json)
     end
 
     context 'with an unacceptable scope in token' do
-      scopes.reject{ |s| s == scope }.each do |invalid_scope|
+      scopes.reject { |s| s == scope }.each do |invalid_scope|
         it "fails for #{invalid_scope} scope" do
           token = Fabricate(:access_token, application: application, scopes: invalid_scope, resource_owner_id: user.id)
-          get url, headers: { 'Authorization': 'Bearer ' + token.token }
-          expect(response).not_to have_http_status(:success)
+          expect do
+            get url, headers: { Authorization: 'Bearer ' + token.token }
+          end.to raise_error(CanCan::AccessDenied)
```

Comment:
On production we redirect to root_path, but in develop and test we just raise the exception:
```ruby
#app/controllers/application_controller.rb

  rescue_from CanCan::AccessDenied do |exception|
    respond_to do |format|
      format.json do
        render json: { status: 403, error: I18n.t('devise.failure.not_permitted_to_view_page') },
               status: 403
      end
      format.all do
        raise exception unless Rails.env.production?
        redirect_to root_path, alert: I18n.t('devise.failure.not_permitted_to_view_page')
      end
    end
  end
```